webpack/assets/javascripts/react_app/components/Template/TemplateInput.js

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import DateTime from '../common/forms/DateTime/DateTime';
+import Autocomplete from './Inputs/AutoComplete';
+
+const TemplateInput = ({
+  data: {
+    label,
+    resourceType,
+    id,
+    description,
+    required,
+    initialError,
+    type,
+    url,
+    supportedTypes,
+    template,
+    value,
+  },
+}) => {
+  const [plain, search, date] = supportedTypes;
+
+  switch (type) {
+    case plain:
+      return null;
+    case search:
+      return (
+        <Autocomplete
+          label={label}
+          id={id}
+          isRequired={required}
+          info={description}
+          initialError={initialError}
+          resourceType={resourceType}
+          url={url}
+          template={template}
+          initialQuery={value}
+        />
+      );
+    case date:
+      return (
+        <DateTime
+          label={label}
+          id={id}
+          isRequired={required}
+          info={description}
+          initialError={initialError}
+          url={url}
+          template={template}
+          hideValue={!value.length}
```

Comment:
This is not going to write the actual value through though, right?
So the user will loose the values on DateTime inputs, or am I missing something?

Reply:
forgot to add the value prop, thanks for noticing !